src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs

```diff
@@ -96,21 +122,49 @@ private unsafe byte[] EncryptOrDecrypt(SafeNCryptKeyHandle key, byte[] input, As
 
             byte[] output = new byte[estimatedSize];
             int numBytesNeeded;
-            ErrorCode errorCode = encryptOrDecrypt(key, input, input.Length, paddingInfo, output, output.Length, out numBytesNeeded, paddingMode);
+            ErrorCode errorCode = encrypt ?
+                Interop.NCrypt.NCryptEncrypt(key, input, input.Length, paddingInfo, output, output.Length, out numBytesNeeded, paddingMode) :
+                Interop.NCrypt.NCryptDecrypt(key, input, input.Length, paddingInfo, output, output.Length, out numBytesNeeded, paddingMode);
+
             if (errorCode == ErrorCode.NTE_BUFFER_TOO_SMALL)
             {
                 output = new byte[numBytesNeeded];
-                errorCode = encryptOrDecrypt(key, input, input.Length, paddingInfo, output, output.Length, out numBytesNeeded, paddingMode);
+                errorCode = encrypt ?
+                    Interop.NCrypt.NCryptEncrypt(key, input, input.Length, paddingInfo, output, output.Length, out numBytesNeeded, paddingMode) :
+                    Interop.NCrypt.NCryptDecrypt(key, input, input.Length, paddingInfo, output, output.Length, out numBytesNeeded, paddingMode);
+
             }
             if (errorCode != ErrorCode.ERROR_SUCCESS)
+            {
                 throw errorCode.ToCryptographicException();
+            }
 
             Array.Resize(ref output, numBytesNeeded);
             return output;
         }
 
-        // Delegate binds to either NCryptEncrypt() or NCryptDecrypt() depending on which api was called.
-        private unsafe delegate ErrorCode EncryptOrDecryptAction(SafeNCryptKeyHandle hKey, byte[] pbInput, int cbInput, void* pPaddingInfo, byte[] pbOutput, int cbOutput, out int pcbResult, AsymmetricPaddingMode dwFlags);
+        // Now that the padding mode and information have been marshaled to their native counterparts, perform the encryption or decryption.
+        private unsafe bool TryEncryptOrDecrypt(
+            SafeNCryptKeyHandle key, ReadOnlySpan<byte> input, Span<byte> output,
+            AsymmetricPaddingMode paddingMode, void* paddingInfo, bool encrypt, out int bytesWritten)
+        {
```

Comment:
I prefer "all on one line", but "multiple lines of multiple" is just a big pile of mush.